charts/workspacemanager/templates/deployment.yaml

```diff
@@ -1,38 +1,104 @@
+{{- $imageTag := .Values.imageConfig.tag | default .Values.global.applicationVersion -}}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ .Chart.Name }}-deployment
+  name: {{ .Values.name }}-deployment
   labels:
-    {{ template "workspacemanager.labels" . }}
+{{ include "workspacemanager.labels" . | indent 4 }}
   {{- if and .Values.proxy.enabled .Values.proxy.reloadOnCertUpdate }}
   annotations:
     reloader.stakater.com/auto: "true"
   {{- end }}
 spec:
+  revisionHistoryLimit: 0
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 0
+      maxUnavailable: 1
   replicas: {{ .Values.replicas }}
   selector:
     matchLabels:
-      app: {{ .Chart.Name }}
+      deployment: {{ .Values.name }}-deployment
   template:
     metadata:
-      name: {{ .Chart.Name }}-service
       labels:
-        app: {{ .Chart.Name }}
+        deployment: {{ .Values.name }}-deployment
+{{ include "workspacemanager.labels" . | indent 8 }}
+      annotations:
+        {{- /* Automatically restart deployments on config map changes: */}}
+        checksum/{{ .Values.name }}-proxy-configmap: {{ include (print $.Template.BasePath "/configmaps/proxy-apache-httpd-configmap.yaml") . | sha256sum }}
+        checksum/{{ .Values.name }}-tcell-configmap: {{ include (print $.Template.BasePath "/configmaps/tcell_configmap.yaml") . | sha256sum }}
+        {{- if .Values.postgres.enabled }}
+        checksum/{{ .Values.name }}-postgres-initdb: {{ include (print $.Template.BasePath "/configmaps/postgres-init-db-configmap.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.prometheus.enabled }}
+        checksum/{{ .Values.name }}-prometheus-configmap: {{ include (print $.Template.BasePath "/configmaps/prometheus-configmap.yaml") . | sha256sum }}
+        {{- end }}
```

Comment:
Could you say more about how this works?
I'm would be very worried about allowing far off config changes to cause WM to rolling restart. Config updates should happen very intentionally per service. If this is referring to a config map that ends up being defined in another "unit of deployment" (today or next quarter), you start to get very mysterious action at a distance, and it's distressing how often config changes bring down services.
I'm ok with this if the restart for deployments are only for changes to the values of the workspace manager helm chart/helm files (and it'd be great if that was enforceable).

Reply:
> Could you say more about how this works?
Sure, I stole it from [here](https://sanderknape.com/2019/03/kubernetes-helm-configmaps-changes-deployments/). Basically it ensures that whenever one of these configmaps is changed as part of a Helm release/ArgoCD sync, the deployment restarts and picks up the changes.
> restart for deployments are only for changes to the values of the workspace manager helm chart/helm files
Yep, this is the case.